src/Robo/Commands/Deploy/DeployCommand.php

```diff
@@ -401,11 +401,19 @@ protected function composerInstall() {
    * Creates deployment_identifier file.
    */
   protected function createDeployId($id) {
-    $this->taskExecStack()->exec("echo '$id' > deployment_identifier")
-      ->dir($this->deployDir)
-      ->stopOnFail()
+    $deployment_identifier_file = $this->getConfigValue('repo.root') . '/deployment_identifier';
```

Comment:
Let's move this logic to a method that is shared so that we don't duplicate code.

Reply:
Where would that live, ideally? Does exposing the `createDeployId` method as a command (maybe `source:build:deployment-identifier`) make sense? Then we could add it to the list of commands invoked in `artifact:build`?

Reply:
That sounds like a good idea!

Reply:
Hmm, `createDeployId` actually seems like it's "on the same level", as `drupal:hash-salt:init`, in a sense. I wonder if it makes more sense to move it into the `drupal` namespace under `drupal:deployment_identifier:init`. The `createDeployId` method is currently invoked from the `drupal:install` command, so it seems like there's definitely an argument that it can fit into the `drupal` namespace, since BLT needs it whenever it installs `drupal`. We could remove it from `drupal:install` and instead add it to the list of commands invoked in `blt setup`:
```
    $commands = [
      'source:build',
      'drupal:hash-salt:init',
      'drupal:deployment-identifier:init`,
    ];
```

and, again, to the list of commands invoked in `artifact:build`:
```
    $this->invokeCommands([
      // Execute `blt source:build:frontend` to ensure that frontend artifact
      // are generated in source repo.
      'source:build:frontend',
      // Execute `drupal:hash-salt:init` to ensure that salt.txt exists.
      // There's a slim chance this has never been generated.
      'drupal:hash-salt:init',
      'drupal:deployment-identifier:init`,
    ]);
```

Thoughts?